agent/inbound.go

```diff
@@ -87,13 +110,36 @@ func (a *Agent) processIncomingAppProject(ev *event.Event) error {
 		return err
 	}
 
+	sourceUIDMatch, err := a.projectManager.CompareSourceUID(a.context, incomingAppProject)
+	if err != nil {
+		return fmt.Errorf("failed to validate source UID of appProject: %w", err)
+	}
+
 	switch ev.Type() {
 	case event.Create:
+		if !sourceUIDMatch {
+			logCtx.Debug("An appProject already exists with a different source UID. Deleting the existing appProject")
+			if err := a.deleteAppProject(incomingAppProject); err != nil {
+				return err
```

Comment:
Same about augmenting the error as above.

Reply:
Updated